### PR TITLE
Added error checks for invalid inputs on thnn_conv2d

### LINKED ISSUE #121188: Segmentation Fault in `torch._C._nn.thnn_conv2d` Due to Invalid Padding Argument
### 🐛 Describe the bug

A segmentation fault occurs when invoking `torch._C._nn.thnn_conv2d` with an empty list for the padding argument, alongside other parameters that are potentially mismatched or improperly specified. 

## Error log

```bash
Segmentation fault (core dumped)
```

## Minimized Repro

```python
import torch
torch._C._nn.thnn_conv2d(torch.rand([1]), bias=torch.rand([3]), kernel_size=[5], padding=[], stride=[9], weight=torch.rand([9]))
```

### Versions

PyTorch version: 2.3.0.dev20240301+cu118
Is debug build: False
CUDA used to build PyTorch: 11.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.3 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.35

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki @malfet

### PR DESCRIPTION
**Pull request by Martim03**
Fixes #121188
Prevent Segmentation Fault in 'torch._C._nn.thnn_conv2d'

Previously, calling 'torch._C._nn.thnn_conv2d' with invalid arguments for padding, stride, and kernel_size would result in a segmentation fault. This issue has been resolved by implementing argument validation (using Torch Check). Now, when invalid arguments are detected, a runtime error is raised with a debug message detailing the correct format.

Additionally, this commit includes tests to cover the three referenced cases.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by janeyx99 — DISMISSED** [review]
This looks good, thanks!

**Comment by Martim03** [comment]
It appears that the tests I made encountered an issue when running with a CUDA build. Regrettably, I only tested them using a CPU build. I will try to obtain a CUDA build in order to investigate if forward. My apologies for the mistake.

**Comment by Martim03** [comment]
After building with cuda, I can confirm that the tests were incorrect. They should be resolved now.

**Comment by Martim03** [comment]
Hi @janeyx99,
the checks were fine on my last commit, but one of them suddenly failed a few weeks ago. Do you have any insights on what might have caused this failure? Apologies for the time gap, I've been quite busy over the past month.

**Comment by janeyx99** [comment]
I am not sure, I don't know of any conv2d changes recently

**Comment by janeyx99** [comment]
Oh, you meant with CI! 

Let's do a rebase and see if it was a failure on main:

@pytorchbot rebase

**Review by janeyx99 — APPROVED** [review]
